src/components/app.jsx

```diff
@@ -1,9 +1,47 @@
-import React from 'react'
+import React from 'react';
+import ViewCards from './view-cards';
+import ReviewCards from './review-cards';
+import CreateCard from './create-card';
+import Nav from './nav';
+
+function View(label, element) {
+  return {
+    label: label,
+    element: element
+  }
+}
 
 export default class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.setView = this.setView.bind(this);
+    this.viewMap = {
+      'view-cards': { label: 'View Grades', element: <ViewCards /> },
+      'review-cards': { label: 'Review', element: <ReviewCards /> },
+      'create-card': { label: 'Create Card', element: <CreateCard /> }
```

Comment:
This is going to become problematic if you need to pass props to these. Typically we do not store React elements in state. Instead, I recommend defining a method that conditionally returns the correct view element based on state.